File: 3418-count-pairs-that-form-a-complete-day-ii/count-pairs-that-form-a-complete-day-ii.py
class Solution:
    def countCompleteDayPairs(self, hours: List[int]) -> int:
        remainder_count = {}
        count_pairs = 0
        
        for hour in hours:
            remainder = hour % 24
            complement = (24 - remainder) % 24  
            
            if complement in remainder_count:
                count_pairs += remainder_count[complement]
            
            if remainder in remainder_count:
                remainder_count[remainder] += 1
            else:
                remainder_count[remainder] = 1
        
        return count_pairs

File: 3418-count-pairs-that-form-a-complete-day-ii/count-pairs-that-form-a-complete-day-ii.java
class Solution {
    public long countCompleteDayPairs(int[] hours) {
        Map<Integer, Integer> remainderCount = new HashMap<>();
        long countPairs = 0;
        for (int hour : hours) {
            int remainder = hour % 24;
            int complement = (24 - remainder) % 24;
            if (remainderCount.containsKey(complement)) {
                countPairs += remainderCount.get(complement);
            }
            if (remainderCount.containsKey(remainder)) {
                remainderCount.put(remainder, remainderCount.get(remainder) + 1);
            } else {
                remainderCount.put(remainder, 1);
            }
        }
        return countPairs;
    }
}
